rts/motoko-rts/src/gc/mark_compact.rs

```diff
@@ -158,15 +158,33 @@ unsafe fn mark_stack<M: Memory>(mem: &mut M, heap_base: u32) {
 }
 
 unsafe fn mark_fields<M: Memory>(mem: &mut M, obj: *mut Obj, obj_tag: Tag, heap_base: u32) {
-    visit_pointer_fields(obj, obj_tag, heap_base as usize, |field_addr| {
-        let field_value = *field_addr;
-        mark_object(mem, field_value);
-
-        // Thread if backwards or self pointer
-        if field_value.get_ptr() <= obj as usize {
-            thread(field_addr);
-        }
-    });
+    visit_pointer_fields(
+        mem,
```

Comment:
Is the mem argument still required? Looks like the old code manage to capture mem in the first callback without it.

Reply:
I tried to beat the borrow check, but it was relentless on me. Only a unified callback would do the trick, and that has other problems.

Reply:
This happens when I close over `mem` from two callbacks: 4db6013188985c37ba8546ef7970e01af9e4bc83.